ModelMetadataExtensions/ConventionalModelMetadataProvider.cs

```diff
@@ -38,16 +38,19 @@ public ConventionalModelMetadataProvider(bool requireConventionAttribute, Type d
 
             var defaultResourceType = DefaultResourceType;
             var conventionAttribute = conventionType.GetAttributeOnTypeOrAssembly<MetadataConventionsAttribute>();
-            if (conventionAttribute != null && conventionAttribute.ResourceType != null)
+            if (conventionAttribute != null)
             {
-                defaultResourceType = conventionAttribute.ResourceType;
+                if (conventionAttribute.ResourceType != null)
+                {
+                    defaultResourceType = conventionAttribute.ResourceType;
+                }
```

Comment:
Yes, I know how the condition evaluated, but please consider that there is an else branch also which cannot be seen on this diff screen... ;)